### DISCUSSION TITLE
[Suggestion] add an option to mount the users home directory as an overlay mount

### DISCUSSION THREAD

**Discussion by slartibart70** [89luca89/distrobox#1904] [Ideas]
**Is your feature request related to a problem? Please describe.**
it's not a problem yet, but when i want to test software (esp. GUI apps) then my homedir will be filled with resource files and other stuff (in ~/.local).
These files must be removed manually after the testing - something not really nice since you don't know what those apps did and where they put stuff (files) or where they modified things in home.

**Describe the solution you'd like**
I could use some overlay-stuff as root, like wrapping my host-homedir with an overlay-fs and using the resulting overlay as an alternative home for distrobox.
But, this needs root (bad for immutable OSes).

podman currently already supports volumes mounted as overlay (`:O,upperdir=/some/upper,workdir=/some/work`) so why not using this for the home-dir when running distrobox?

If this was exported as distrobox-option (e.g. 'immutable-home=true') with the provisioning of the necessary 'upper' and 'work' dirs, then we would have a bit more isolation, esp. when testing/trying out unknown apps inside distrobox

Any chance to see this in a future release?

**Comment by 89luca89** [comment]
You can also use `--home` to point $HOME somewhere else and let the test app litter that directory

**Comment by slartibart70** [comment]
yes, i know of this.

But the point is, i want to have the files of HOME and additionally a protection to not mess with them... as an overlay does
--home does not provide this, because this goes into a separate dir

hence the idea with overlays

**Comment by 89luca89** [comment]
adding `--home` does not remove access to your original home :+1:

**Comment by GavinNL** [comment]
Hi, I was looking for a feature like this as well. 

My main reason, like OP said is to run an application with full access to my HOME folder, but also have the container protect it from modifications. If we have an overlay HOME, then we can also see what the container modified by viewing the mounted directory on the host.

The --home doesn't quite do what I would like.

**Comment by realyukii** [comment]
I see on the docs that `---home` is only available for `create`, but not available for `enter` :"
